### ISSUE TITLE
Regression in 1.8.0: Line with formatting immediately following line with HTML doesn't render

### ISSUE THREAD

**Issue by lukasbestle** [erusev/parsedown#721]
Hi,

One of our Kirby customers reported a bug in Parsedown 1.8.0-beta7 (see https://github.com/getkirby/kirby/issues/1943):

**Markdown input:**

```markdown
<p>Simple HTML</p>
# Test
```

**Output from [Parsedown 1.7.3](https://parsedown.org/demo):**

```html
<p>Simple HTML</p>
<h1>Test</h1>
```

**Output from Parsedown 1.8.0-beta7:**

```html
<p>Simple HTML</p>
# Test
```

**Code example:**

```php
$markdownWithHtml = <<<MARKDOWN
<p>Simple HTML</p>
# Test
MARKDOWN;

$parsedown = new Parsedown();
echo $parsedown->text($markdownWithHtml);
```

**Comment by aidantwoods** [comment]
Hi there, this is intended behaviour and [replicated in the reference parser](https://spec.commonmark.org/dingus/?text=%3Cp%3ESimple%20HTML%3C%2Fp%3E%0A%23%20Test).

Reasoning for this is that the line `<p>Simple HTML</p>` starts a [HTML block](https://spec.commonmark.org/0.29/#html-blocks) (of type 6 in this case), which requires a blank line to end it per the CommonMark spec. This seems a bit weird at first, but (AFAIK) this is because CommonMark tries to avoid having to parse potentially more complicated HTML in a correct way—instead it roughly says that if something looks like it starts a block of HTML then treat it like HTML until the next line break (there are a few special cases where this differs, but this is the gist). This generally works well, but can sometimes be surprising (as you point out here).

**Comment by aidantwoods** [comment]
Basically, the way around this treatment of HTML blocks (and generally with CommonMark things) is to separate separate things by blank lines where possible, i.e. write:

```markdown
<p>Simple HTML</p>

# Test
```

**Comment by aidantwoods** [comment]
Looking at the referenced issue though, this *is* a bug when the HTML block of concern is `script`, `pre`, or `style` however. This is fixed in the `2.0.x` branch but not yet in `1.8.x-beta`

**Comment by lukasbestle** [comment]
Thanks for the quick reply! Is there any timeframe for a release of the `2.0.x` branch? Or if not, would you recommend using the current state of the `2.0.x` branch over `1.8.0-beta7` in production?

**Comment by schulzrinne** [comment]
This is a problem even if the post-HTML text is separated by a blank line. For example, pasting the following after the demo text leaves the "Regards" part unparsed. Both <h4> and <p> are block-level elements and there's a blank line between the </p> and the "Regards".

<h4 >Overall score: How do you score this paper?</h4><p >weak accept (3)</p>

Regards,  
The Conference Chairs & TPC

**Comment by aidantwoods** [comment]
@schulzrinne this issue was focussed on 1.8-beta (the demo page still uses 1.7 afaik) does this happen in the latest release of the beta branch?

**Comment by schulzrinne** [comment]
Yes, 1.8.0-beta-7 appears to fix this issue. Thanks for the quick follow up.

**Comment by lukasbestle** [comment]
@aidantwoods The original issue I've reported is not resolved (I tested that with `1.8.0-beta7` as I wrote in the original post). Could you please reopen the issue?
